utilities-core/src/main/java/uk/gov/justice/services/common/converter/jackson/additionalproperties/AdditionalPropertiesDeserializer.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.justice.services.common.converter.jackson.additionalproperties;
+
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.BeanDeserializer;
+import com.fasterxml.jackson.databind.deser.BeanDeserializerBase;
+import org.apache.commons.lang3.reflect.FieldUtils;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.ParameterizedType;
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.justice.services.common.converter.jackson.additionalproperties.AdditionalPropertiesHelper.ADDITIONAL_PROPERTIES_NAME;
+
+
+public class AdditionalPropertiesDeserializer extends BeanDeserializer {
+
+    public AdditionalPropertiesDeserializer(final BeanDeserializerBase originalDeserializer) {
+        super(originalDeserializer);
+    }
+
+    @Override
+    public void handleUnknownVanilla(final JsonParser jsonParser,final DeserializationContext context, final Object bean, final String propName) throws IOException {
+
+        final Object value = deserializeFromType(jsonParser, context, jsonParser.getCurrentToken());
+
+        Map<String, Object> additionalPropertiesMap;
+
+        try {
+            final Field additionalPropertiesMapField = FieldUtils.getDeclaredField(bean.getClass(), ADDITIONAL_PROPERTIES_NAME, true);
+
+            if (isValidAdditionalPropertiesField(additionalPropertiesMapField)) {
+
+                additionalPropertiesMap = (Map<String, Object>) FieldUtils.readDeclaredField(bean, ADDITIONAL_PROPERTIES_NAME, true);
```

Comment:
Need to consider the possibility it's an immutable map returned from `readDeclaredField`